### DISCUSSION TITLE
Support for Pandoc's (Haskell Core's) Table of Contents Write Option

### DISCUSSION THREAD

**Discussion by simonvincenthk** [boisgera/pandoc#64] [Q&A]
Hi,

I'd like to use `pandoc`'s (the haskell core's) `--table-of-contents=true`/`--toc=true` [general writer option](https://pandoc.org/MANUAL.html#general-writer-options) to automatically generate a table of contents when writing a `Pandoc` object to a markdown file. 

Every time I pass `--table-of-contents=true` or `--toc=true`  as an `option` to the [`.write()` method](https://boisgera.github.io/pandoc/api/), I get an error. 

Accordingly, I'm wondering if this option is supported and if I need to do anything special to use it?

I'm using
* `pandoc` (haskell core) [version 2.19.2](https://github.com/jgm/pandoc/releases/tag/2.19.2), and 
* `pandoc` (python wrapper) [version  2.3](https://boisgera.github.io/pandoc/changelog/)

Best regards, Simon

**Comment by boisgera** [comment]
Hi @simonvincenthk!

The code 

```python
import pandoc

md = """

# Header 1

## Header 1.1

### Header 1.2

"""

doc = pandoc.read(md)

md_out = pandoc.write(doc, format="markdown", options=["-s", "--toc"])

print(md_out)
```

should output

```markdown 
-   [Header 1](#header-1)
    -   [Header 1.1](#header-1.1)
        -   [Header 1.2](#header-1.2)

Header 1
========

Header 1.1
----------

### Header 1.2
```

(except that ATM, I am running into #65 for some recent versions of pandoc(/Haskell) ; with 2.9.x I think that you should be fine)

What error do you get?

Best regards,

Sébastien
